R/import_helpers.R

```diff
@@ -22,7 +22,7 @@ long_to_ts <- function(data) {
   dt_of_lists <- data_dt[, {
     dT <- diff(date_zoo)
     if(any(diff(dT) > 1e-6)) {
-      list(ts_object = list(xts(value, order_by = date_zoo)))
+      list(ts_object = list(xts(value, order.by = as.yearmon(date_zoo))))
```

Comment:
can we always use this? or do we need as.yearqtr as well?

Reply:
Actually I'm not even sure we need `as.yearmon` here. date_zoo should already be either yearmon or yearqtr by line 25 (lines 19-20). Need to try and find a suitable test case.

We only use xts in case of irregular time series anyway `any(diff(dT) > 1e-6)`, otherwise we return a ts (line 27).

Reply:
... except we essentially unclass them with `as.numeric`.